### PR TITLE
Added max_multi_level_up battle config

### PR DESCRIPTION
**Pull request by kaninhot004**
* **Addressed Issue(s)**: multi_level_up battle config have no maximum settings.

* **Server Mode**: Both

* **Description of Pull Request**: Add cap the Level, Job Level that increase by leveling.

### CODE REVIEW

**Comment by rAthenaAPI** [comment]
This pull request **introduces 1 alert** when merging e8c2dfeaa7511d4fecc4db70b39fd0c903f83004 into 92ba773dee78dd72ce93cd1fdc2fab33da5ae8a4 - [view on LGTM.com](https://lgtm.com/projects/g/rathena/rathena/rev/pr-6e30d1313371002a06e6f883f44e5cc894c2d17b)

**new alerts:**

* 1 for Dead code due to goto or break statement

---

**Warning** - Automated code review for rathena/rathena will be disabled on October 1, 2019. You can avoid this by <a href="https://github.com/apps/lgtm-com/installations/new">installing the LGTM.com GitHub App</a>. Read about the <a href="https://lgtm.com/blog/announcing_github_apps_migration">benefits of migrating to GitHub Apps</a> in the blog.

---

*Comment posted by [LGTM.com](https://lgtm.com)*

File: src/map/battle.cpp [hunk 1]
```diff
@@ -8257,7 +8257,8 @@ static const struct _battle_data {
 	{ "pk_level_range",                     &battle_config.pk_level_range,                  0,      0,      INT_MAX,        },
 	{ "manner_system",                      &battle_config.manner_system,                   0xFFF,  0,      0xFFF,          },
 	{ "pet_equip_required",                 &battle_config.pet_equip_required,              0,      0,      1,              },
-	{ "multi_level_up",                     &battle_config.multi_level_up,                  0,      0,      1,              },
+	{ "max_multi_level_up_base",            &battle_config.max_multi_level_up_base,         0,      0,      MAX_LEVEL,      },
+	{ "max_multi_level_up_job",             &battle_config.max_multi_level_up_job,          0,      0,      MAX_LEVEL,      },
```
Comment (secretdataz) [inline src/map/battle.cpp]: Default value in exp.conf is 1 but here is 0.

File: src/map/pc.cpp [hunk 2]
```diff
@@ -6800,10 +6800,11 @@ int pc_checkbaselevelup(struct map_session_data *sd) {
 	if (!next || sd->status.base_exp < next || pc_is_maxbaselv(sd))
 		return 0;
 
+	int maxMultiLevelUp = cap_value(battle_config.max_multi_level_up_base, 1, MAX_LEVEL);
```
Comment (secretdataz) [inline src/map/pc.cpp]: Perhaps make this variable snake_case?

File: src/map/pc.cpp [hunk 3]
```diff
@@ -6869,10 +6873,12 @@ int pc_checkjoblevelup(struct map_session_data *sd)
 	if(!next || sd->status.job_exp < next || pc_is_maxjoblv(sd))
 		return 0;
 
+	
+	int maxMultiLevelUp = cap_value(battle_config.max_multi_level_up_job, 1, MAX_LEVEL);
```
Comment (secretdataz) [inline src/map/pc.cpp]: same as above

**Comment by kaninhot004** [comment]
@secretdataz All fixed from your suggestion and default value mistake. Thank you!

**Comment by kaninhot004** [comment]
> From a quick glance at the code, it doesn't seem like you can level up and have some exp leftover. If you're at 90 exp, you level up at 100 exp, and you gain 20 exp, you should be at 10exp. Can you test this?

I will try EXP leftover. I've set it to 0 because my friends show me up 
image

I will set the EXP as much as it can be leftover.

Thank you!

File: src/map/pc.cpp [hunk 4]
```diff
@@ -6800,10 +6800,11 @@ int pc_checkbaselevelup(struct map_session_data *sd) {
 	if (!next || sd->status.base_exp < next || pc_is_maxbaselv(sd))
 		return 0;
 
+	int max_multi_level_up = cap_value(battle_config.max_multi_level_up_base, 1, MAX_LEVEL);
```
[the remark this answers is not in the document]
Comment (kaninhot004) [reply src/map/pc.cpp]: Thank you again!

File: src/map/pc.cpp [hunk 5]
```diff
@@ -6800,10 +6800,11 @@ int pc_checkbaselevelup(struct map_session_data *sd) {
...
 	do {
 		sd->status.base_exp -= next;
 		//Kyoki pointed out that the max overcarry exp is the exp needed for the previous level -1. [Skotlex]
-		if( ( !battle_config.multi_level_up || ( battle_config.multi_level_up_base > 0 && sd->status.base_level >= battle_config.multi_level_up_base ) ) && sd->status.base_exp > next-1 )
+		if( ( ( battle_config.multi_level_up_base > 0 && sd->status.base_level >= battle_config.multi_level_up_base ) ) && sd->status.base_exp > next-1 )
```
[the remark this answers is not in the document]
Comment (kaninhot004) [reply src/map/pc.cpp]: Fix, Thank you!

**Comment by kaninhot004** [comment]
@vstumpf This is proof when using current changes.

Cap at 1
At Level 1 gain 2700 EXP (4900%) >> Lv 2 (0%)
Lv1ToLv2Cap1

At Level 2 gain 27 EXP (30%) >> Lv 1 (30%)
Lv2Cap1

Cap at 175
At Level 1 gain gain 2700 EXP (4900%) >> Lv 10 (18.3%)
Lv1ToLv10Cap175

=================================

So if you want left over EXP that result will be

Cap at 1
At Level 1 gain 2700 EXP (4900%) >> Lv 2 (99%)

I think it's not cool. But if you want just tell me.

Thanks!

**Pull request closed without merging**